d3servercheck.ps1

```diff
@@ -1,70 +1,52 @@
-function Write-Server-Result($tcpconns) {
+function Write-ServerResult($tcpconns) {
     if (!$tcpconns) {
         return
     }
-
-    foreach ($tcpconn in $tcpconns) {
-        Write-Host "$($tcpconn.RemoteAddress) (" -NoNewline
-        if ($is_good_ip.ContainsKey($tcpconn.RemoteAddress)) {
-            if ($is_good_ip[$tcpconn.RemoteAddress]) {
-                Write-Host "GOOD" -NoNewline
+    $Unique_Addresses = $tcpconns | Sort-object -property RemoteAddress -unique
+    write-host "Connections:"
+    foreach ($address in $Unique_Addresses) {
+        if ($is_good_ip.ContainsKey($address.RemoteAddress)) {
+            if ($is_good_ip[$address.RemoteAddress]) {
+                $Result = "GOOD"
             } else {
-                Write-Host "BAD" -NoNewline
+                $Result =  "BAD" 
             }
-        } else {
-            Write-Host "UNKNOWN" -NoNewline
         }
-
-        Write-Host ") " -NoNewline
+        else{
+            $Result = 'Unknown'
+        }
+        
+        $Message = '{0} : ({1})' -f $address.remoteaddress,$Result
+        Write-Host "$message"
     }
+    write-host "`r"
 }
 
 Write-Host "Starting d3servercheck"
 
 $csv = Import-Csv -Path servers.csv
 
+try {
+    $Diablo3_Process = get-process -name 'Diablo III64' -ErrorAction Stop
```

Comment:
I don't want the script to fail if D3 is not running, as this is annoying when restarting D3 (e.g. due to party invitation bugs etc.). I'd rather move this to the `while`-loop and just print a short message like "no running D3 game detected" instead.